### PR TITLE
Add subgraph GitHub workflow and update the Readme

### PR DESCRIPTION
**Pull request by dimitrovmaksim**
Signed-off-by: Maksim Dimitrov <dimitrov.maksim@gmail.com>

1. Adds a GitHub workflow for the subgraph indexing tests
2. Updates the readme with introduction to TheGraph, How to run a private node against `testnet`, `previewnet` and `mainnet` 

**Related issue(s)**:

Fixes https://github.com/hashgraph/hedera-json-rpc-relay/issues/709
Fixes https://github.com/hashgraph/hedera-json-rpc-relay/issues/772

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by georgi-l95 — DISMISSED** [review]
LGTM

File: tools/subgraph-example/README.md [hunk 1]
```diff
@@ -10,29 +12,35 @@ GravatarRegistry -> GravatarCreated and GravatarUpdated
...
+
+1. [What is TheGraph](https://thegraph.com/docs/en/about)
+2. [How to create a subgraph](https://thegraph.com/docs/en/developing/creating-a-subgraph/)
+3. [The AssemblyScript API](https://thegraph.com/docs/en/developing/assemblyscript-api/)
+
+## Try it out:
+
+### Prerequisites:
 
 Note: Currently this example needs to be executed against older relay and mirror-node versions, until all fixes have been released.
```
Comment (lukelee-sl) [inline tools/subgraph-example/README.md:25]: can we specify which versions work?

Comment (dimitrovmaksim) [reply tools/subgraph-example/README.md]: Updated the readme and added a link to the full custom local-node config

Comment (lukelee-sl) [reply tools/subgraph-example/README.md]: Nice!  Thank you

**Review by lukelee-sl — APPROVED** [review]
LGTM

**Review by georgi-l95 — APPROVED** [review]
LG

**Pull request merged**